release-tools/stochss_ami_manager.py

```diff
@@ -0,0 +1,547 @@
+        pass
+
+    def __cleanup_instance(self):
+        header = 'Cleaning up crumbs...'
+        print '=================================================='
+        print header
+        commands = ['sudo rm -f /etc/ssh/ssh_host_*',
+                    'sudo rm -f ~/.ssh/authorized_keys',
+                    'sudo rm -f ~/.bash_history']
+
+        command = ';'.join(commands)
+        self.__run_remote_command(command=command, log_header=header)
+
+    def make_image(self):
+        print '=================================================='
+        print 'Making AMI...'
+        date_string = time.strftime("%Y%b%d-%H%M%S")
+        new_ami_name = "StochSS-Node-{0}".format(date_string)
+
+        print "Creating AMI '{0}' from instance {1}...".format(new_ami_name, self.instance_id)
+        new_ami_id = self.ec2_connection.create_image(self.instance_id, new_ami_name)
+
+        print "New AMI pending..."
+        new_ami = self.ec2_connection.get_image(new_ami_id)
+
+        while new_ami.state != 'available':
+            time.sleep(5)
+            new_ami.update()
+
+        print "New AMI ID: " + new_ami_id
+        print "Making the new AMI Public..."
+        self.ec2_connection.modify_image_attribute(new_ami_id,
+                                                   attribute='launchPermission',
+                                                   operation='add',
+                                                   groups='all')
+
+
+    def __terminate_instance(self):
+        print '=================================================='
+        print 'Terminating launched instance...'
+        self.ec2_connection.terminate_instances(instance_ids=[self.instance_id])
+        print 'Done.'
+
+    def __check_dependency_installation(self):
+        header = 'Checking dependencies...'
+        print '=================================================='
+        print header
+        expected_dependency_list_string = ' '.join(self.dependencies)
+        command = "dpkg-query -W -f='\${Package}\\n' " + expected_dependency_list_string
+
+        tmp_log_files = {
+            "stdout": tempfile.TemporaryFile(),
+            "stderr": tempfile.TemporaryFile()
+        }
+
+        self.__run_remote_command(command=command, log_files=tmp_log_files, silent=True)
+        tmp_log_files["stdout"].seek(0)
+
+        dependency_list_string = tmp_log_files["stdout"].read().strip()
+
+        tmp_log_files["stdout"].close()
+        tmp_log_files["stderr"].close()
+
+        if sorted(dependency_list_string.split('\n')) == sorted(self.dependencies):
+            print 'All dependencies are installed!'
+            return True
+
+        else:
+            print 'Some dependencies are missing ...'
+            print 'List of installed dependencies: \n{0}\n'.format(dependency_list_string)
+            return False
+
+    def __check_fenics_installation(self):
+        # TODO: check FeniCS installation
+        pass
```

Comment:
This would be very useful, since the image should not be created without a working fenics.  And their install has been buggy previously.  The check is as simple as checking the return code of the command `python -c "import dolfin" 2>/dev/null`